plugins/dev-tools/src/common_test_helpers.mocha.js

```diff
@@ -5,54 +5,65 @@
  */
 
 /**
- * Test case configuration information.
+ * Test case configuration.
  * @record
  */
-export function TestCase() {}
-/**
- * @type {string} The title for the test case.
- */
-TestCase.prototype.title = '';
-/**
- * @type {boolean|undefined} Whether this test case should be skipped. Used to
- *    skip buggy test case and should have an associated bug.
- */
-TestCase.prototype.skip = false;
-/**
- * @type {boolean|undefined} Whether this test case should be called as only.
- *    Used for debugging.
- */
-TestCase.prototype.only = false;
+export class TestCase {
+  /**
+   * Class for a test case configuration.
+   */
+  constructor() {
+    /**
+     * @type {string} The title for the test case.
+     */
+    this.title;
+    /**
+     * @type {boolean|undefined} Whether this test case should be skipped.
+     *   Used to skip buggy test case and should have an associated bug.
+     */
+    this.skip;
+    /**
+     * @type {boolean|undefined} Whether this test case should be called as
+     *   only. Used for debugging.
+     */
+    this.only;
+  }
+}
 
 /**
- * Test suite configuration information.
+ * Test suite configuration.
  * @record
  * @template {TestCase} T
- * @template {TestSuite} U
  */
-export function TestSuite() {}
-/**
- * @type {string} The title for the test case.
- */
-TestSuite.prototype.title = '';
-/**
- * @type {boolean} Whether this test suite should be skipped. Used to
- *    skip buggy test case and should have an associated bug.
- */
-TestSuite.prototype.skip = false;
-/**
- * @type {boolean} Whether this test suite should be called as only.
- *    Used for debugging.s
- */
-TestSuite.prototype.only = false;
-/**
- * @type {?Array<T>} The associated test cases.
- */
-TestSuite.prototype.testCases = [];
-/**
- * @type {?Array<U>} Optional inner test suites.
- */
-TestSuite.prototype.testSuites = [];
+export class TestSuite {
+  /**
+   * Class for a test suite configuration.
+   */
+  constructor() {
+    /**
+     * @type {string} The title for the test case.
+     */
+    this.title;
+    /**
+     * @type {?Array<T>} The associated test cases.
+     */
+    this.testCases;
+    /**
+     * @type {?Array<TestCase<T>>} Optional inner test suites.
```

Comment:
I think the previous annotation is correct, because test suites can now be parameterized with their inner test suite type. In Java this would look like:
```java
public interface TestSuite<U extends TestSuite, T extends TestCase> { }

// Sort of a recursive type definition, but totally legal.
public class GeneratorTestSuite implements TestSuite<GeneratorTestSuite, GeneratorTestCase> { }
```

If we declare it as `?Array<!TestSuite<T>>` I don't that's not correct because the subclass of `TestSuite` (eg `GeneratorTestSuite`) could have properties that don't exist on `TestSuite` and this type annotation would make those properties illegal to access, because it would be legal to store any type of `TestSuite` including the base one, in this array.

Reply:
That's a good point.

The problem I'm trying to solve is that `runTestSuites` currently makes the assumption that the `testSuites` stored accept the same `TestCase` type (`T`) as the parent `TestSuite`, so I'm trying to more formally capture that relationship.

I have a revised version, but I need to do some digging on if it's correct.

Reply:
Hmmm I wonder if it's possible to declare something like:
```java
// Added <U, T> to the TestSuite type parameter
public interface TestSuite<U extends TestSuite<U, T>, T extends TestCase> { }
```

I think that notates the fact that the inner `TestSuite` type must recursively contain itself, and the same `TestCase` type as the outer `TestSuite`.

But I'm not sure if the closure compiler supports type constraints like that.

Reply:
I've reverted my changes to the class annotation and updated the method `runTestSuites`

Reply:
Looks correct to me :D